packages/core/src/client.ts

```diff
@@ -266,9 +267,10 @@ export interface TokenResponse {
   readonly profile: Reference<ProfileResource>;
 }
 
-export interface BotEvent {
+export interface BotEvent<T = Resource | Hl7Message | string> {
   readonly contentType: string;
-  readonly input: Resource | Hl7Message | string;
+  readonly input: T;
+  readonly secrets?: ProjectSecret[];
```

Comment:
To simplify the Bot code, can we index these by name before sending them to the bot? E.g make secrets a `Record<string, ProjectSecret>` and non-optional

**Before** 
```ts
if(!secrets){ throw new Error() ;}
const mySecret = secrets.find((e) => e.name === 'MySecret')l
if(!mySecret) {throw new Error();} 
```
** After **
```ts
const mySecret = secrets['MySecret`]
if(!mySecret) { throw new Error() }
```


We can still store the secrets as an array in the Resource, but just index them in the wrapper code when the Bot is invoked

Reply:
Yeah, I like that